### PR TITLE
Update Android and iOS SDK

### PR DESCRIPTION
**Pull request by moritzmoritz**
Closes https://github.com/Driversnote-Dev/react-native-helpscout-beacon/issues/19

* Update iOS and Android SDK versions
* Include Beacon as dependency in Podspecs

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,15 @@ _The format is based on [Keep a Changelog](http://keepachangelog.com/) and this
 
 ## [unreleased]
 
+## [v0.6.0] - 2024-09-18
```
Comment (martinpoulsen) [inline CHANGELOG.md:9]: This version change entry should be postponed to when we create the release - just leave it under `unreleased` for now

File: package.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "react-native-helpscout-beacon",
   "title": "React Native Helpscout Beacon",
-  "version": "0.5.0",
+  "version": "0.6.0",
   "description": "Integrate Helpscout Beacon in to your React Native app",
-  "main": "lib/index.js",
-  "types": "lib/types.d.ts",
```
Comment (martinpoulsen) [inline package.json:7]: Are you sure you should remove the `lib`? Did you change anything to warrant this change? It looks to me like the package might stop working if the consumer of the package doesn't transpile the TS themselves.

**Review by martinpoulsen — COMMENTED** [review]
The SDK updates look good 👍

I'm mostly curious about the changes to distribute the non-compiled TypeScript files.

File: package.json (hunk 2 above)
Comment (moritzmoritz) [reply package.json:7]: We don't have a `lib` folder in the package present so this is pointing now to the right file. :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,13 @@ _The format is based on [Keep a Changelog](http://keepachangelog.com/) and this
 
 ## [unreleased]
 
+### Changed
+
+- Android native SDK is updated to v. 5.1.2
+- iOS native SDK is updated to v. 3.0.1
+- `package.json` is now pointing to the right index and types file
```
Comment (martinpoulsen) [inline CHANGELOG.md:13]: I think this entry should be reverted now

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react-native-helpscout-beacon",
   "title": "React Native Helpscout Beacon",
-  "version": "0.5.0",
+  "version": "0.6.0",
```
Comment (martinpoulsen) [inline package.json:4]: Sorry, missed this in my last review, the version bump should be part of the release version commit

File: android/build.gradle [hunk 5]
```diff
@@ -106,7 +106,7 @@ def configureReactNativePom(def pom) {
 
 afterEvaluate { project ->
     task androidSourcesJar(type: Jar) {
-        classifier = 'sources'
+        archiveClassifier = 'sources'
```
Comment (martinpoulsen) [inline android/build.gradle:109]: 👍

**Review by martinpoulsen — APPROVED** [review]
Looking great now 💯

**Pull request merged**